frontend/components/commons/sidebar-feedback-task/SidebarFeedbackTask.component.vue

```diff
@@ -1,21 +1,115 @@
+  <div class="sidebar__container">
+    <SidebarFeedbackTaskPanel
+      :visible-panel="currentMetric"
+      @close-panel="closePanel"
+    >
+      <component v-if="getProgressComponentName" :is="getProgressComponentName"
+    /></SidebarFeedbackTaskPanel>
+    <SidebarFeedbackTask
+      @on-click-sidebar-action="onClickSidebarAction"
+      :sidebar-items="sidebarItems"
+      :active-buttons="[currentMetric, currentMode]"
+      :expanded-component="currentMetric"
+    />
+  </div>
 </template>
 
 <script>
-export default {};
+export default {
+  data: () => ({
+    currentMetric: null,
+    currentMode: "annotate",
+  }),
+  computed: {
+    getProgressComponentName() {
+      return (
+        this.sidebarItems.find((item) => item.id === this.currentMetric)
+          ?.component || null
+      );
+    },
+  },
+  created() {
+    this.sidebarItems = [
+      {
+        id: "annotate",
+        tooltip: "Hand labeling",
+        icon: "hand-labeling",
+        group: "Mode",
+        action: "change-view-mode",
+        type: "non-expandable",
+        relatedMetrics: ["progress", "stats"],
+      },
+      {
+        id: "explore",
+        tooltip: "Exploration",
+        icon: "exploration",
+        group: "Mode",
+        action: "change-view-mode",
+        type: "non-expandable",
+        relatedMetrics: ["progress", "stats"],
+      },
+      {
+        id: "progress",
+        tooltip: "Progress",
+        icon: "progress",
+        action: "show-metrics",
+        group: "Metrics",
+        type: "expandable",
+        component: "FeedbackTaskProgress",
+      },
+      {
+        id: "stats",
+        tooltip: "Stats",
+        icon: "stats",
+        action: "show-metrics",
+        type: "expandable",
+        group: "Metrics",
+      },
+      {
+        id: "refresh",
+        tooltip: "Refresh",
+        icon: "refresh",
+        group: "Refresh",
+        type: "non-expandable",
+        action: "refresh",
+      },
+    ];
+  },
+  methods: {
+    onClickSidebarAction(action, info) {
+      switch (action) {
+        case "show-metrics":
+          this.showMetrics(info);
+          break;
+        case "change-view-mode":
+          console.log("change-view-mode", info);
+          break;
+        case "refresh":
+          console.log("refresh dataset");
+          break;
+        default:
+          console.warn(action);
+      }
+    },
+    showMetrics(info) {
+      this.currentMetric = this.currentMetric !== info ? info : null;
+    },
+    closePanel() {
+      this.currentMetric = null;
+    },
+  },
+};
 </script>
 
 <style lang="scss" scoped>
 .sidebar {
-  display: flex;
-  flex-direction: column;
-  width: 100%;
-  height: 100%;
-  gap: 1em;
-  border-left: 1px solid #e6e6e6;
-  background: #f5f5f5;
-  pointer-events: all;
-  transition: box-shadow 0.2s ease-in-out 0.4s;
+  &__container {
+    z-index: 1;
+    position: fixed;
+    display: flex;
+    right: 0;
+    pointer-events: none;
+  }
```

Comment:
order : from more important to less important => display, position/right, z-index , pointer-events